include/odp/api/spec/ipsec.h

```diff
@@ -1226,12 +1226,23 @@ typedef struct odp_ipsec_status_t {
  * e.g. RFC 4302 and 4303). Resulting packets are well formed, reconstructed
  * original IP packets, with IPSEC headers removed and valid header field values
  * restored. The amount and content of packet data before the IP header is
- * undefined.
+ * undefined. Some amount of TFC padding may follow the IP packet payload,
+ * in which case packet length is larger than protocol headers indicate.
+ * TFC dummy packets have l3_type set to ODP_PROTO_L3_TYPE_NONE in tunnel mode
```

Comment:
It also should be NO_NEXT, shouldn't it?

Reply:
L3 types are basically what you find from L2 type field: ARP, RARP, IPV4, IPV6. There's no "no next header" type in e.g. ethernet types. So, NO_NEXT is defined only on L4 types. But that's OK since, for tunnel mode application just checks agaist l3_type == IPV4, IPV6 or NONE, where NONE == dummy. In transport mode, application just checks l4_type == NO_NEXT for dummy. Application L3 type in transport should always have either IPV4 or IPV6.